services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,994 @@
+        flow.allocateProtectedPath = true
+        flowHelper.addFlow(flow)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not create flow: Couldn't setup protected path for one-switch flow"
+    }
+
+    def "Unable to update flow with the 'protected path' option on a single switch flow"() {
+        given: "A switch"
+        def sw = topology.activeSwitches.first()
+
+        and: "A flow without protected path"
+        def flow = flowHelper.singleSwitchFlow(sw)
+        flowHelper.addFlow(flow)
+
+        when: "Update flow: enable protected path"
+        northbound.updateFlow(flow.id, flow.tap { it.allocateProtectedPath = true })
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not update flow: Couldn't setup protected path for one-switch flow"
+
+        and: "Cleanup: revert system to original state"
+        flowHelper.deleteFlow(flow.id)
+    }
+
+    @Unroll
+    def "System is able to reroute #flowDescription flow to protected path"() {
+        given: "Two active not neighboring switches with two possible paths at least"
+        def allIsls = northbound.getAllLinks()
+        def (Switch srcSwitch, Switch dstSwitch) = getNotNeighboringSwitchPair(2)
+
+        when: "Create flow with protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = bandwidth
+        flow.allocateProtectedPath = true
+        flowHelper.addFlow(flow)
+
+        then: "Flow is created with protected path"
+        def flowPathInfo = northbound.getFlowPath(flow.id)
+        flowPathInfo.protectedPath
+        Wrappers.wait(WAIT_OFFSET) { assert northbound.getFlowStatus(flow.id).status == FlowState.UP }
+
+        and: "Current path is not equal to protected path"
+        def currentPath = pathHelper.convert(flowPathInfo)
+        def currentProtectedPath = pathHelper.convert(flowPathInfo.protectedPath)
+        currentPath != currentProtectedPath
+
+        and: "Bandwidth is reserved for protected path on involved isls"
+        def protectedIsls = pathHelper.getInvolvedIsls(currentProtectedPath)
+        def protectedIslsInfo = protectedIsls.collect { islUtils.getIslInfo(it).get() }
+
+        allIsls.each { originIsl ->
+            protectedIslsInfo.each { newIsl ->
+                if (originIsl.id == newIsl.id) {
+                    assert originIsl.availableBandwidth - newIsl.availableBandwidth == flow.maximumBandwidth
+                }
+            }
+        }
+
+        when: "Init reroute"
+        def currentIsls = pathHelper.getInvolvedIsls(currentPath)
+        def islToBreak = currentIsls.find { !protectedIsls.contains(it) }
```

Comment:
It looks like we do not need to do it because current ISLs never should intersect with protected ISLs.

Reply:
I think we should leave it as is
I guess we can have this situation when flow goes through A-switch